src/Raven.Studio/typescript/common/queryCompleter.ts

```diff
@@ -0,0 +1,279 @@
+/// <reference path="../../typings/tsd.d.ts" />
+
+import collectionsTracker = require("common/helpers/database/collectionsTracker");
+import getIndexTermsCommand = require("commands/database/index/getIndexTermsCommand");
+import getDocumentsMetadataByIDPrefixCommand = require("commands/database/documents/getDocumentsMetadataByIDPrefixCommand");
+import database = require("models/resources/database");
+import getIndexEntriesFieldsCommand = require("commands/database/index/getIndexEntriesFieldsCommand");
+import collection = require("models/database/documents/collection");
+import document = require("models/database/documents/document");
+
+class queryCompleter {
+    private tokenIterator: new(session : AceAjax.IEditSession, initialRow: number, initialColumn: number) => AceAjax.TokenIterator = ace.require("ace/token_iterator").TokenIterator;
+    private collectionsTracker: collectionsTracker;
+    private indexFieldsCache = new Map<string, string[]>();
+    private defaultScore = 10000;
+    
+    constructor(private activeDatabase: KnockoutObservable<database>,
+                private indexes: KnockoutObservableArray<Raven.Client.Documents.Operations.IndexInformation>) {
+        this.collectionsTracker = collectionsTracker.default;
+    }
+    
+    private getIndexName(session: AceAjax.IEditSession): [string, boolean] {
+        let keyword: string;
+        
+        for (let row = 0; row < session.getLength(); row++) {
+            let lineTokens: AceAjax.TokenInfo[] = session.getTokens(row);
+
+            for (let i = 0; i < lineTokens.length; i++) {
+                const token = lineTokens[i];
+                switch (token.type) {
+                    case "keyword": {
+                        keyword = token.value.toLowerCase();
+                        break;
+                    }
+                    case "string": {
+                        const indexName = token.value.substr(1, token.value.length - 2);
```

Comment:
No, as string would always have the ' or " or ` at the beginning and at the end.